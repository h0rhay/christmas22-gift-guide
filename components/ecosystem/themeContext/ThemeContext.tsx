import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
