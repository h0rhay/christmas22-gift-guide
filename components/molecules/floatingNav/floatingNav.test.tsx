import { render, screen } from '@testing-library/react';
import FloatingNav from '.';
import { ThemeProvider } from 'styled-components';
import Theme from 'styles/theme';
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";

const mockCategoryList = [
  {
    id: 0,
    slug: 'foo'
  },
  {
    id: 1,
    slug: 'bar'
  },
  {
    id: 2,
    slug: 'foobar'
  }
]

let renderComponent: () => { container: any; };

describe('Floating Nav', () => {

  describe('when selected category is neither first nor last category', () => {
    beforeEach(() => {
      renderComponent = () =>
        render(
          <ThemeProvider theme={Theme}>
            <ThemeContext colorGroup="hero">
              <FloatingNav categoryId={1} categories={mockCategoryList} />
            </ThemeContext>
          </ThemeProvider>
        );
    });

    it('should match snapshot', () => {
      const { container } = renderComponent();
      expect(container).toMatchSnapshot('selected category is neither first nor last category');
    });

    it('should have home, prev and next button', () => {
      renderComponent();
      const home = screen.getByText(/home/i);
      const prev = screen.getByText(/prev/i);
      const next = screen.getByText(/next/i);
      expect(home).toBeInTheDocument();
      expect(prev).toBeInTheDocument();
      expect(next).toBeInTheDocument();
    });

    it('should have two svg buttons', () => {
      const { container } = renderComponent();
      container.querySelectorAll('svg').length.valueOf(2);
    });
  });

  describe('when first category is selected', () => {
    beforeEach(() => {
      renderComponent = () =>
        render(
          <ThemeProvider theme={Theme}>
            <ThemeContext colorGroup="hero">
              <FloatingNav categoryId={0} categories={mockCategoryList} />
            </ThemeContext>
          </ThemeProvider>
        );
    });

    it('should match snapshot', () => {
      const { container } = renderComponent();
      expect(container).toMatchSnapshot('first category is selected');
    });

    it('should have home and next button & not prev button', () => {
      renderComponent();
      const home = screen.getByText(/home/i);
      const prev = screen.queryByText(/prev/i);
      const next = screen.getByText(/next/i);
      expect(home).toBeInTheDocument();
      expect(prev).not.toBeInTheDocument();
      expect(next).toBeInTheDocument();
    });

  });

  describe('when last category is selected', () => {
    beforeEach(() => {
      renderComponent = () =>
        render(
          <ThemeProvider theme={Theme}>
            <ThemeContext colorGroup="hero">
              <FloatingNav categoryId={2} categories={mockCategoryList} />
            </ThemeContext>
          </ThemeProvider>
        );
    });

    it('should match snapshot', () => {
      const { container } = renderComponent();
      expect(container).toMatchSnapshot('last category is selected');
    });

    it('should have home and prev button & not next button', () => {
      renderComponent();
      const home = screen.getByText(/home/i);
      const prev = screen.getByText(/prev/i);
      const next = screen.queryByText(/next/i);
      expect(home).toBeInTheDocument();
      expect(prev).toBeInTheDocument();
      expect(next).not.toBeInTheDocument();
    });

  });

  describe('when no category is selected', () => {
    beforeEach(() => {
      renderComponent = () =>
        render(
          <ThemeProvider theme={Theme}>
            <ThemeContext colorGroup="hero">
              <FloatingNav categoryId={-1} categories={mockCategoryList} />
            </ThemeContext>
          </ThemeProvider>
        );
    });

    it('should match snapshot', () => {
      const { container } = renderComponent();
      expect(container).toMatchSnapshot('no category is selected');
    });

    it('should have only next button', () => {
      renderComponent();
      const home = screen.queryByText(/home/i);
      const prev = screen.queryByText(/prev/i);
      const next = screen.getByText(/next/i);
      expect(home).not.toBeInTheDocument();
      expect(prev).not.toBeInTheDocument();
      expect(next).toBeInTheDocument();
    });

  });

});
