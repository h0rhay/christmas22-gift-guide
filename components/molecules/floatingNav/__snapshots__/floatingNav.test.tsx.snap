// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floating Nav when first category is selected should match snapshot: first category is selected 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="ThemeContext__ColoredSurface-sc-7o1m9v-0 rMDNb colored-surface"
    >
      <div
        class="floatingNav__Wrapper-sc-16mop7q-0 gecmAm"
        data-projection-id="13"
        data-test-id="floatingNav"
        draggable="false"
        style="user-select: none;"
      >
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="14"
        >
          <span>
            HOME
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="15"
        >
          <span>
            NEXT
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="16"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="17"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
            />
            <polyline
              points="16 6 12 2 8 6"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="15"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Floating Nav when last category is selected should match snapshot: last category is selected 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="ThemeContext__ColoredSurface-sc-7o1m9v-0 rMDNb colored-surface"
    >
      <div
        class="floatingNav__Wrapper-sc-16mop7q-0 gecmAm"
        data-projection-id="23"
        data-test-id="floatingNav"
        draggable="false"
        style="user-select: none;"
      >
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="24"
        >
          <span>
            HOME
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="25"
        >
          <span>
            PREV
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="26"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="27"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
            />
            <polyline
              points="16 6 12 2 8 6"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="15"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Floating Nav when no category is selected should match snapshot: no category is selected 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="ThemeContext__ColoredSurface-sc-7o1m9v-0 rMDNb colored-surface"
    >
      <div
        class="floatingNav__Wrapper-sc-16mop7q-0 gecmAm"
        data-projection-id="33"
        data-test-id="floatingNav"
        draggable="false"
        style="user-select: none;"
      >
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="34"
        >
          <span>
            NEXT
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="35"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
          data-projection-id="36"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
            />
            <polyline
              points="16 6 12 2 8 6"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="15"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Floating Nav when selected category is neither first nor last category should match snapshot: selected category is neither first nor last category 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="ThemeContext__ColoredSurface-sc-7o1m9v-0 rMDNb colored-surface"
    >
      <div
        class="floatingNav__Wrapper-sc-16mop7q-0 gecmAm"
        data-test-id="floatingNav"
        draggable="false"
        style="user-select: none;"
      >
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
        >
          <span>
            HOME
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
        >
          <span>
            PREV
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
        >
          <span>
            NEXT
          </span>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </button>
        <button
          class="circleButton__Wrapper-sc-o294s7-0 kmASHX"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
            />
            <polyline
              points="16 6 12 2 8 6"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="15"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
